programs/sbf/tests/programs.rs

```diff
@@ -4112,41 +4112,56 @@ fn test_cpi_invalid_account_info_pointers() {
     let mut bank_client = BankClient::new_shared(bank);
     let authority_keypair = Keypair::new();
 
-    let (_bank, c_invoke_program_id) = load_program_of_loader_v4(
-        &mut bank_client,
-        &bank_forks,
-        &mint_keypair,
-        &authority_keypair,
-        "invoke",
-    );
-    let (bank, invoke_program_id) = load_program_of_loader_v4(
-        &mut bank_client,
-        &bank_forks,
-        &mint_keypair,
-        &authority_keypair,
-        "solana_sbf_rust_invoke",
-    );
-
     let account_keypair = Keypair::new();
     let mint_pubkey = mint_keypair.pubkey();
-    let account_metas = vec![
+    let mut account_metas = vec![
         AccountMeta::new(mint_pubkey, true),
         AccountMeta::new(account_keypair.pubkey(), false),
-        AccountMeta::new_readonly(invoke_program_id, false),
-        AccountMeta::new_readonly(c_invoke_program_id, false),
     ];
 
-    for invoke_program_id in [invoke_program_id, c_invoke_program_id] {
+    let mut program_ids: Vec<Pubkey> = Vec::with_capacity(2);
+
+    #[cfg(feature = "sbf_rust")]
+    {
+        let (_bank, invoke_program_id) = load_program_of_loader_v4(
+            &mut bank_client,
+            &bank_forks,
+            &mint_keypair,
+            &authority_keypair,
+            "solana_sbf_rust_invoke",
+        );
+        account_metas.push(AccountMeta::new_readonly(invoke_program_id, false));
+        program_ids.push(invoke_program_id);
+    }
+
+    #[cfg(feature = "sbf_c")]
+    {
+        let (_bank, c_invoke_program_id) = load_program_of_loader_v4(
+            &mut bank_client,
+            &bank_forks,
+            &mint_keypair,
+            &authority_keypair,
+            "invoke",
+        );
+        account_metas.push(AccountMeta::new_readonly(c_invoke_program_id, false));
+        program_ids.push(c_invoke_program_id);
+    }
+
+    let bank = bank_client
```

Comment:
This should have happened in load_program_of_loader_v4(). YOu just need to overwrite the bank variable.